netmaster/master/network.go

```diff
@@ -236,11 +276,18 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		nwCfg.ExtPktTag = int(extPktTag)
 		nwCfg.PktTag = int(pktTag)
 	} else if network.PktTagType == "vxlan" {
+		if !IsTagInRange(network.PktTag, "vxlan", gCfg.Auto.VXLANs, stateDriver, tenantName) {
+			return fmt.Errorf("vxlan %d does not adhere to tenant's vxlan range %s", network.PktTag, gCfg.Auto.VXLANs)
+		}
+
 		nwCfg.ExtPktTag = network.PktTag
 		nwCfg.PktTag = network.PktTag
```

Comment:
@vvb - this line is still a problem.
The changes you made are needed, however not sufficient if we want to allow vxlan range beyond 4k. for example:
```
                } else if nwCfg.PktTagType == "vxlan" {
                        extPktTag, pktTag, err = gCfg.AllocVXLAN(rm)
                        if err != nil {
                                return err
                        }
                }
```
The following code in auto-allocation allows for both vlan-tag and vxlan-tag to be assigned; we need to follow the similar model for the code above to avoid a conflict and accessing beyond the range.

Bottom line: if you try a vxlan id beyond 4k range, it might still not work even with all your changes. In order to fix this you'd have to allocate a vlan-tag for it that is still in the allowed range. If you follow ofnet.AddVlan() call from ovsdriver.go for CreateNetwork() may be it will all make sense.

I understand you had not added all this, but let's fix this all of them.